plugins/video/pkg_src/src/decoder/video_decoder_mixed.cc

```diff
@@ -71,28 +73,18 @@ void VideoDecoderMixed::Run(dali::Workspace &ws) {
   int batch_size = input.num_samples();
   int s = 0;
 
-  cuInit(0);
-  int nGpu = 0;
-  cuDeviceGetCount(&nGpu);
 
-  bool m_bDestroyContext = false;
   CUcontext cuContext = nullptr;
-  CUstream cuStream = nullptr;
+  CUstream cuStream = ws.stream();
+  cuCtxGetCurrent(&cuContext);
 
-  for (int i = 0; i < batch_size; i++) {
-    cuCtxGetCurrent(&cuContext);
-    if (!cuContext) {
-      createCudaContext(&cuContext, device_id_, 0);
-      m_bDestroyContext = true;
-    }
-    cuCtxPopCurrent(&cuContext);
+  if (!cuContext) {
+    throw std::runtime_error("Failed to create a cuda context");
   }
 
-  if (!cuContext)
-    throw std::runtime_error("Failed to create a cuda context");
+  auto output_sample = output[s];
```

Comment:
shouldn't this go inside the loop? Right now you seem to be writing to the first sample output buffer always

Reply:
Done